src/Controllers/PushController.js

```diff
@@ -58,7 +60,55 @@ export class PushController extends AdaptableController {
     body['expiration_time'] = PushController.getExpirationTime(body);
     // TODO: If the req can pass the checking, we return immediately instead of waiting
     // pushes to be sent. We probably change this behaviour in the future.
-    rest.find(config, auth, '_Installation', where).then(function(response) {
+    let badgeUpdate = Promise.resolve();
+
+    if (body.badge) {
```

Comment:
You actually don't need to use this piece, since this happens before PushAdapter, which is the extension point where SNS would be plugged in.